// Copyright 2017 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// -----------------------------------------------------------------------------

syntax = "proto3";

option java_multiple_files = true;
option java_package = "sawtooth.config.protobuf";

message ValidatorInfo {
    string name = 1;            // 验证者名字

    string id = 2;              // 验证者公钥

    bytes public_key = 3;      // 通过验证者私钥生成的 vrf 公钥

    bytes previous_id = 4;     // 前一个区块的 id

    double expectation = 5;     // 验证者的期望

    bytes proof = 6;           // vrf证明

    bytes proof_data = 7;      // vrf验证的消息的字节串

    bytes random = 8;          // vrf生成的随机数
}
